### PR TITLE
⬆️ Upload illustratory customization of information assemblages

### LINKED ISSUE #170: Allow a custom photo to be uploaded for a resource
There should be an optional field in the admin form to upload a specific photo for a resource.  This would entail making a change in the resource model to store a link (?) and changing the resource preview to display the custom photo.

https://ant.design/components/upload/

### PR DESCRIPTION
**Pull request by Feeshay**
Resolves #170 

- Allows admins to upload and remove custom images
- Upon upload, modal appears to allow admins to crop to 7:5 aspect ratio
- After cropping, image is stored in backend as B64 string
- Custom images appear in cards
- Input validation (uploaded file must be a jpg/png smaller than 2mb)

Codepath:
Upload image with antd Upload component
Crop image to 7:5 aspect ratio 
Encode the cropped image into a B64 string
Call endpoint to save the string into database
Call endpoint to retrieve string from database
Display image in cards with B64 string source

References:
Cropper - https://github.com/ricardo-ch/react-easy-crop - image selected by user
Uploading - https://ant.design/components/upload/ - photo wall, avatar
Slider - https://ant.design/components/slider/

Pictures:
image
image

### CODE REVIEW

File: client/src/utils/api.js [hunk 1]
```diff
@@ -3,7 +3,7 @@
 const axios = require('axios');
 
 const instance = axios.create({
-  baseURL: 'https://ymca.now.sh',
+  baseURL: 'http://localhost:9000',
```
Comment (n3a9) [inline client/src/utils/api.js:6]: For production we should definitely have this as ymca.now.sh :)

File: client/src/components/ResourceImageUpload.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    const reader = new FileReader();
+    // setImage to B64 string upon successful image read
+    reader.addEventListener('load', () => setCroppingImg(reader.result));
+    // send image to reader
+    reader.readAsDataURL(event.file.originFileObj);
+    setShowCropper(true);
+  };
+
+  return (
+    <div>
```
Comment (n3a9) [inline client/src/components/ResourceImageUpload.js:89]: `<div>` -> `<>`

File: client/src/components/ResourceImageUpload.js [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+  const beforeUpload = file => {
+    const isValidImage =
+      file.type === 'image/jpeg' || file.type === 'image/png';
+    if (!isValidImage) {
+      message.error('You can only upload JPG/PNG file!');
+    }
+    const isSmall = file.size / 1024 / 1024 < 2;
+    if (!isSmall) {
+      message.error('Image must smaller than 2MB!');
```
Comment (n3a9) [inline client/src/components/ResourceImageUpload.js:73]: In the backend I think we set this to 5MB, so we should either make it 2 or 5. I think either is fine

File: client/src/utils/api.js (hunk 1 above)
Comment (jackieo5023) [inline client/src/utils/api.js:6]: Make sure to switch this back :)

**Comment by Feeshay** [comment]
The image has to be <2 mb but that isn't counting everything else (so the entire json could be a bit more than 2 mb). I'll make it something closer to 2 mb though

**Pull request merged**